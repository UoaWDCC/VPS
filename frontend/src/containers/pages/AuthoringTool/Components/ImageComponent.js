/* eslint-disable no-param-reassign */
import React, { useEffect, useState } from "react";
import Image from "material-ui-image";
import { useGet } from "../../../../hooks/crudHooks";
import { imageStyles, imageContainerStyles } from "./components.styles";

export default function ImageComponent({ id, onClick, component }) {
  const [image, setImage] = useState();
  useGet(`/api/image/${component.imageId}`, setImage, false);

  useEffect(() => {
    // The below is used to recalculate the height and width of an image. Especially when it is intiallised using auto
    const imageElement = document.getElementById(id);
    const canvas = document.getElementById("canvas")?.getBoundingClientRect();
    if (canvas && component && imageElement) {
      component.width = (imageElement.offsetWidth / canvas.width) * 100;
      component.height = (imageElement.offsetHeight / canvas.height) * 100;
    }
  }, [image]);

  return (
    <Image
      id={id}
      src={image ? image.url : ""}
      imageStyle={imageStyles(component)}
      style={imageContainerStyles()}
      onClick={onClick}
    />
  );
}
