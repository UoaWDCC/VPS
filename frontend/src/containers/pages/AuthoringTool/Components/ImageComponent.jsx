/* eslint-disable no-param-reassign */
import Image from "material-ui-image";
import { useEffect, useState } from "react";
import { useGet } from "../../../../hooks/crudHooks";
import { imageContainerStyles, imageStyles } from "./components.styles";

/**
 * This component represents an image scene component
 * @component
 * @example
 * <ImageComponent
 *    id={index}
 *    onClick={onClick}
 *    component={component}
 * />
 */
export default function ImageComponent({ id, onClick, component }) {
  const [image, setImage] = useState();
  useGet(`/api/image/${component.imageId}`, setImage, false);

  useEffect(() => {
    // The below is used to recalculate the height and width of an image. Especially when it is intiallised using auto
    const imageElement = document.getElementById(id);
    const canvas = document.getElementById("canvas")?.getBoundingClientRect();
    if (canvas && component && imageElement) {
      component.width = (imageElement.offsetWidth / canvas.width) * 100;
      component.height = (imageElement.offsetHeight / canvas.height) * 100;
    }
  }, [image]);

  return (
    <Image
      id={id}
      src={image ? image.url : ""}
      imageStyle={imageStyles(component)}
      style={{
        ...imageContainerStyles(),
        zIndex: component?.zPosition || 0,
      }}
      onClick={onClick}
    />
  );
}
