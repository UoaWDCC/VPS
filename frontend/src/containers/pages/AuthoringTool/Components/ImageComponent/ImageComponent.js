import React, { useState } from "react";
import Image from "material-ui-image";
import { useGet } from "../../../../../hooks/crudHooks";
import { imageStyles, imageContainerStyles } from "../components.styles";

export default function ImageComponent({ id, selectElement, component }) {
  const [image, setImage] = useState();
  useGet(`/api/image/${component.imageId}`, setImage);

  return (
    <Image
      id={id}
      src={image ? image.url : ""}
      imageStyle={imageStyles(component)}
      style={imageContainerStyles(component)}
      onClick={selectElement}
    />
  );
}
