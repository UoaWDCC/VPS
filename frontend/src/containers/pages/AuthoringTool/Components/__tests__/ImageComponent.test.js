import React from "react";
import { render } from "@testing-library/react";
import ImageComponent from "../ImageComponent";

test("Add image component snapshot test", () => {
  const { baseElement } = render(<ImageComponent component={{}} />);

  expect(baseElement).toMatchSnapshot();
});
