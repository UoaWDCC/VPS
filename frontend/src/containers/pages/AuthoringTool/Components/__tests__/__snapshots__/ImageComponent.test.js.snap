// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add image component snapshot test 1`] = `
<body>
  <div>
    <div
      style="background-color: rgb(255, 255, 255); position: relative; display: contents; cursor: pointer;"
    >
      
      <div
        style="width: 100%; height: 100%; position: absolute; top: 0px; left: 0px; display: flex; align-items: center; justify-content: center; pointer-events: none;"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 48px; height: 48px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;
